File: bootcamp_c_cpp_dsa-main/bootcamp_c_cpp_dsa-main/C/2darry/p10.c
// a to p


#include<stdio.h>
void main(){

	int x=10;
	int y=20;
	int z=30;

	int *arr[3]={&x,&y,&z};
	printf("%ld\n",arr);
	printf("%d\n",arr[0]);
	printf("%d\n",arr[1]);
	printf("%d\n",arr[2]);

	printf("%d\n",*(arr[0]));
	printf("%d\n",*(arr[1]));
	printf("%d\n",*(arr[2]));

}

File: java-main/java-main/AE/p10.java
import java.util.HashMap;
class FirstRepeatingElement {
    public static void main(String[] args) {
        int[] arr = {1, 5, 3, 4, 3, 5, 6};
        int n = arr.length;
        
        HashMap<Integer, Integer> elementIndices = new HashMap<>();

        int firstRepeatingIndex = -1;

        for (int i = 0; i < n; i++) {
            int currentElement = arr[i];

                        if (elementIndices.containsKey(currentElement)) {
                
                if (firstRepeatingIndex == -1 || elementIndices.get(currentElement) < firstRepeatingIndex) {
                    firstRepeatingIndex = elementIndices.get(currentElement);
                }
            } else {
                
                elementIndices.put(currentElement, i);
            }
        }

        if (firstRepeatingIndex != -1) {
		System.out.println("First Repeating Element Index: " + (firstRepeatingIndex + 1));
        } else {
            System.out.println("No repeating elements found.");
        }
    }
}

